packages/jsii-spec/lib/spec.ts

```diff
@@ -324,40 +412,53 @@ export enum PrimitiveType {
     Json = 'json',
 
     /**
-     * Value with "any" or "unknown" type (aka Object)
+     * Value with "any" or "unknown" type (aka Object). Values typed `any` may
+     * be `null` or `undefined`.
      */
     Any = 'any'
 }
 
 /**
  * A reference to a type (primitive, collection or fqn).
  */
-export type TypeReference = TypeReferenceBase & (NamedTypeReference | PrimitiveTypeReference | CollectionTypeReference | UnionTypeReference);
+export type TypeReference = NamedTypeReference | PrimitiveTypeReference | CollectionTypeReference | UnionTypeReference;
 
 /**
- * Common attributes of a TypeReference.
+ * The standard representation of the `any` type (includes optionality marker).
  */
-export interface TypeReferenceBase {
+export const CANONICAL_ANY: Readonly<PrimitiveTypeReference> = { optional: true, primitive: PrimitiveType.Any };
```

Comment:
Why does `any` imply `optional`? It implies nullable, but that's a different thing.

Reply:
I guess I think of optional as allowing "no value" through, and any also allows "no value" though. But I guess I see your point still...